checker/src/abstract_value.rs

```diff
@@ -1186,7 +1186,7 @@ impl AbstractValueTrait for Rc<AbstractValue> {
                 default.try_to_retype_as(target_type),
             ),
             Expression::TaggedExpression { operand, tag } => {
-                operand.try_to_retype_as(target_type).add_tag(*tag)
+                operand.try_to_retype_as(target_type).add_tag(tag.clone())
```

Comment:
clone is normally an indication of an expensive operation. Perhaps not appropriate here?

Reply:
The tag type now contains Rc<String> so it cannot be copied. Anyway add_tag needs a copy/clone to record the tag in the tag domain.